File: Cpp/backjoon/gold/18119.cpp
#include <iostream>
#include <string>

using namespace std;

int arr[100001] = {};

int main() {
	int n, m;
	int mask;
	int o, x;
	for (int i = 0; i < 26; i++) {
		mask |= 1 << i;
	}

	cin >> n >> m;
	string temp;
	for (int i = 0; i < n; i++) {
		cin >> temp;
		for (auto j: temp) {
			arr[i] |= 1 << (j - 'a');
		}
	}

	for (int i = 0; i < m; i++) {
		cin >> o >> x;
		int count = 0;
		mask ^= 1 << (x - 'a');
		for (auto j: arr) {
			if (j == (j & mask)) count++;
		}
		cout << count;
	}

	return 0;
}

File: Python/backjoon/gold/18119.py
import sys

input = sys.stdin.readline

a = ord('a')


def make_ord(chr):
    return ord(chr) - a


mask = 0
for i in range(26):
    mask |= 1 << i
# print(ords)
# print(bin(mask))
N, M = map(int, input().split(' '))
N_arr = [0 for i in range(N)]
for i in range(N):
    str_ = input().rstrip()
    bit = 0
    for j in str_:
        bit |= 1 << (make_ord(j))
    N_arr[i] = bit

for i in range(M):
    o, x = input().rstrip().split(' ')
    count = 0
    mask ^= 1 << make_ord(x)
    for j in N_arr:
        if j == (j & mask):
            count += 1

    print(count)
#     print(N_arr)
#     print(N_first)
#     answer.append(count)
# print(answer)
# print(N_arr)
# print(N_first)
